src/test/unit/analysis/test_plugin_base.py

```diff
@@ -150,33 +113,30 @@ def test_no_multithread(self):
     def test_normal_multithread(self):
         self.multithread_config_test(False, '2', '2')
 
-    @mock.patch('plugins.base.ViewUpdater', lambda *_: None)
     def test_init_result_dict(self):
-        self.p_base = DummyPlugin(self, self.config)
+        self.p_base = DummyPlugin(self.MockPluginAdministrator())
         resultdict = self.p_base.init_dict()
-        self.assertIn('analysis_date', resultdict, 'analysis date missing')
-        self.assertEqual(resultdict['plugin_version'], '0.0', 'plugin version field not correct')
+        assert 'analysis_date' in resultdict, 'analysis date missing'
+        assert resultdict['plugin_version'] == '0.0', 'plugin version field not correct'
         self.p_base.shutdown()
 
 
-class TestPluginTimeout(TestPluginBase):
+@pytest.mark.AnalysisPluginClass.with_args(DummyPlugin)
+@pytest.mark.plugin_start_worker
+@pytest.mark.skip('TODO somehow setting analysis_plugin.TIMEOUT has no effect on AnalysisBasePlugin.TIMEOUT')
```

Comment:
It would be good if we could fix this before merging

Reply:
Fixed. The fix revealed sth. what is in my opinion a design flaw in PluginBase.
My plans to fix this reside in https://github.com/fkie-cad/FACT_core/pull/809/commits/fc90d07b0568644768b7235b14150a826da73820 of #809. But since this will probably take some time before this can be merged I added a workaround and documented it.